components/bastion/setup.ftl

```diff
@@ -0,0 +1,173 @@
+  )]
+
+  [@createNetworkInterface
+    id=resources["networkInterface"].Id
+    name=resources["networkInterface"].Name
+    location=regionId
+    nsgId=getReference(
+      networkResources["networkSecurityGroup"].Id,
+      networkResources["networkSecurityGroup"].Name)
+    ipConfigurations=[nicIpConfiguration]
+    dependsOn=
+      [
+        getReference(resources["publicIP"].Id, resources["publicIP"].Name)
+      ]
+  /]
+
+  [#-- VM Scale Set --]
+  [#local vmssProcessorProfile = processors[solution.Profiles.Processor]]
+  [#local vmssProcessorType = vmssProcessorProfile[core.Type]]
+  [#local vmssProcessor = vmssProcessorType.Processor]
+  [#local vmssProcessorTier = vmssProcessor?split("_")[0]]
+  [#local vmssVMImageProfile = vmImageProfiles[BASTION_COMPONENT_TYPE]]
+  [#local vmssVMAdminName = BASTION_COMPONENT_TYPE]
+
+  [#if deploymentSubsetRequired("parameters", false)]
+      [@addParametersToDefaultJsonOutput
+        id=baselineComponentIds["ssh"]
+        parameter=getKeyVaultParameter(
+          baselineLinks[0].State.Resources["keyVault"].Id, 
+          baselineComponentIds["ssh"]
+        )
+      /]
+  [/#if]
+  [#local vmssVMOSConfig = getVirtualMachineProfileLinuxConfig(
+    [
+      getVirtualMachineProfileLinuxConfigPublicKey(
+        "/home/" + vmssVMAdminName + "/.ssh_authorized_keys",
+        getParameterReference(baselineComponentIds["ssh"])
+      )
+    ],
+    true
+  )]
+
+  [#local vmssVMProfile = getVirtualMachineProfile(
+    core.Type,
+    vmssVMAdminName,
+    "Standard_LRS",
+    vmssVMImageProfile.Publisher,
+    vmssVMImageProfile.Offering,
+    vmssVMImageProfile.SKU,
+    [{ 
+      "name" : resources["networkInterface"].Name,
+      "id" : getReference(
+      resources["networkInterface"].Id,
+      resources["networkInterface"].Name) 
+    }],
+    vmssVMOSConfig
+  )]
+
+  [@createVMScaleSet
+    id=resources["scaleSet"].Id
+    identity={"type": "SystemAssigned"}
+    name=resources["scaleSet"].Name
+    location=regionId
+    skuName=vmssProcessor
+    skuTier=vmssProcessorTier
+    skuCapacity=autoScaleConfig.MinUpdateInstances
+    vmProfile=vmssVMProfile
+  /]
+  
+
+  [#-- AutoScale Policy --]
+  [#if autoScaleConfig.Enabled]
+
+    [#local autoScaleTargetId = getReference(
+        resources["scaleSet"].Id,
+        resources["scaleSet"].Name)]
+
+    [#-- 
+      This autoScaleRule reviews the sum of network bandwidth over a period
+      of time, and if the machine is found to be inactive, scales down by 1.
+      When applied to the autoScaleProfile (with max count of 1 and min of 0)
+      this effectively means a bastion host only exists for the period it is
+      in use.
+    --]
+    [#local autoScaleRule = getAutoScaleRule(
+      "Network In Total",
+      autoScaleTargetId,
+      "PT1M",
+      "Sum",
+      "PT5M",
+      "Average",
+      "LessThan",
+      50000,
+      "Decrease",
+      "ChangeCount",
+      "PT30M",
+      1)]
```

Comment:
I'm not sure network bandwidth is a safe way to determine usage of a VM. An ssh session looking at logs can be quite small, especially if there are agents also reporting off to Azure and what not during that time

Reply:
As discussed, 3e08cea implements a manual intervention approach for scaling for the bastion MVP. We'll revisit autoscaling for it in time.